### PR TITLE
Refactor MultibuildPackage concern

### PR DESCRIPTION
**Pull request by vpereira**
* Move code from accessing the `_multibuild` file from the concern to the backend lib
* Caching the file content for subsequent queries
* Add specs

### CODE REVIEW

File: src/api/app/lib/backend/api/sources/package.rb [hunk 1]
```diff
@@ -51,6 +51,18 @@ def self.service(project_name, package_name)
           nil
         end
 
+        # Returns the content of the _multibuild file (if present)
+        # @return [String]
+        def self.multibuild(project_name, package_name)
+          Rails.cache.fetch("multibuild_#{project_name}_#{package_name}") do
```
Comment (hennevogel) [inline src/api/app/lib/backend/api/sources/package.rb:57]: A cache is nice but a cache also need cache invalidation.

In this case the file is controlled by the backend so the backend would need to invalidate the cache. I wouldn't have an idea how...

Comment (vpereira) [reply src/api/app/lib/backend/api/sources/package.rb:57]: `"multibuild_#{cache_key_with_version}"` would be enough no? The content of this file won't touch without the package container (package model) being changed, right?

Comment (vpereira) [reply src/api/app/lib/backend/api/sources/package.rb:57]: I don't think the `_multibuild` files changes so often to justify a query to backend everytime we call 
` Package.get_by_project_and_name(target_project_name, target_package_name, follow_multibuild: true)`

Comment (hennevogel) [reply src/api/app/lib/backend/api/sources/package.rb:57]: > The content of this file won't change without the package container (package model) being changed, right?

Not right. The backend will change this file at will.

**Pull request closed without merging**